fffw/encoding/filters.py

```diff
@@ -278,8 +278,18 @@ def transform(self, *metadata: Meta) -> Meta:
                 scenes.append(Scene(stream=scene.stream, start=start,
                                     duration=end - start))
 
-        return replace(meta, start=self.start, duration=self.end,
-                       scenes=scenes, streams=streams)
+        kwargs = {
+            'start': self.start,
+            'duration': self.end,
+            'scenes': scenes,
+            'streams': streams
+        }
+        interval = cast(TS, self.end) - cast(TS, self.start)
+        if isinstance(meta, AudioMeta):
+            kwargs['samples'] = round(meta.sampling_rate * interval)
+        if isinstance(meta, VideoMeta):
+            kwargs['frames'] = round(meta.frame_rate * interval)
```

Comment:
Is it normal that 10.1 will be rounded to 10, but 10.9 will become 11?

Reply:
Well, mediainfo sometimes uses INTEGER SECONDS to compute frames count, so it's not a big deal.
All metadata transformations are used to simplify unit-testing and debugging, so it's more or less about internal ffmpeg compute model. And in `validate()` method `+-1` delta is allowed.